### ISSUE TITLE
Relax click version upper bound

### ISSUE THREAD

**Issue by szeleeteo** [Kludex/uvicorn#1016]
Uvicorn can only use [click](https://click.palletsprojects.com/en/7.x/) package version 7.* as defined in `setup.py`

``` python
minimal_requirements = [
    "click==7.*",
    ...
]
```

However, there is currently a pre-release version [8.0.0a1](https://pypi.org/project/click/8.0.0a1/) which fixes many [issues](https://github.com/pallets/click/blob/master/CHANGES.rst#version-80)

Would like to request for uvicorn to relax the upper bound for click requirement, such that user can optionally use click version 8 if applicable
``` python
minimal_requirements = [
    "click>=7.*"
    ...
]
```

**Comment by mariobriggs** [comment]
opened a PR

**Comment by epogrebnyak** [comment]
> However, there is currently a pre-release version [8.0.0a1](https://pypi.org/project/click/8.0.0a1/) which fixes many [issues](https://github.com/pallets/click/blob/master/CHANGES.rst#version-80)

Just to mention for the purposes of #1033 - Click 8.0.0 is a major release since May 11, so locking at 7.* needs an update.

**Comment by epogrebnyak** [comment]
My temp solution is `poetry add click=7.*`

**Comment by mariobriggs** [comment]
click v8.0.0 released 2 days ago

**Comment by simonw** [comment]
This is blocking me from upgrading [Datasette](https://github.com/simonw/datasette) to Click 8.0, since Datasette has `uvicorn` as a dependency.

**Comment by euri10** [comment]
closed by #1042
